### PR TITLE
Make a workaround for Spring boot run as jar

### LINKED ISSUE #8705: Favicon not shown with JAR packaging
### Description of the bug
When packaging a project as a JAR, using a custom `icon.png` in `src/main/resources/META-INF/resources/icons`, and running with `java -jar ...`, the custom favicon is not shown (it is shown correctly when running the `main` method in IntelliJ IDEA, though). The problem is the same with both development and production builds.
### Minimal reproducible example
Go to https://vaadin.com/start/v14. Create a Spring Boot project. Replace the `icon.png` file with a custom one. Build with `mvn package`. Run with ´java -jar ...`.
### Expected behavior
Should show the custom icon.
### Actual behavior
Shows the default white circle with a blue background icon.
### Versions:
    - Vaadin / Flow version: 14.3.0
    - Java version: 8 and 11
    - OS version: macOS
    - Browser version (if applicable): Chrome

### PR DESCRIPTION
**Pull request by denis-anisimov**
Fixes #8705

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/PwaRegistry.java [hunk 1]
```diff
@@ -127,6 +132,19 @@ public PwaRegistry(PWA pwa, ServletContext servletContext)
         }
     }
 
+    private URL getUrl(ServletContext context, String path)
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/PwaRegistry.java:135]: I suggest to improve the naming a bit: `getUrl` -> `getResourceUrl`, `URL logo` -> `resourceUrl`

File: flow-server/src/main/java/com/vaadin/flow/server/PwaRegistry.java [hunk 2]
```diff
@@ -96,18 +99,20 @@ public PwaRegistry(PWA pwa, ServletContext servletContext)
 
         // Build pwa elements only if they are enabled
         if (pwaConfiguration.isEnabled()) {
-            URL logo = servletContext
-                    .getResource(pwaConfiguration.relIconPath());
-            URL offlinePage = servletContext
-                    .getResource(pwaConfiguration.relOfflinePath());
+            URL logo = getUrl(servletContext, pwaConfiguration.relIconPath());
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/PwaRegistry.java:102]: Does it make sense to cover this by a test? Just to be sure this workaround won't be removed accidentally.

Comment (denis-anisimov) [reply flow-server/src/main/java/com/vaadin/flow/server/PwaRegistry.java]: For many reasons I didn't want to make a test.
But let it be ... 
The test makes impossible to test getting icon resource from servlet context - that was the primary reason.

File: flow-server/src/test/java/com/vaadin/flow/server/PwaRegistryTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+public class PwaRegistryTest {
+
+    @Test
+    public void pwaIconIsGeneratedBasedOnClasspathIcon_servletContextHasNoResources()
+            throws IOException {
+        ServletContext context = Mockito.mock(ServletContext.class);
+        PwaRegistry registry = new PwaRegistry(
+                PwaRegistryTest.class.getAnnotation(PWA.class), context);
+        List<PwaIcon> icons = registry.getIcons();
+        PwaIcon pwaIcon = icons.stream().filter(icon -> icon.getWidth() == 32)
```
Comment (mshabarov) [inline flow-server/src/test/java/com/vaadin/flow/server/PwaRegistryTest.java:38]: Add a comment that this filtered `pwaIcon` is the custom one, and got from `META-INF/resources/icons/icon.png`

**Pull request merged**